ui/src/components/app/app-router.tsx

```diff
@@ -16,19 +18,27 @@ import {
 
 export const appRouter = createHashRouter(
   createRoutesFromElements(
-    <Route element={<MainLayout />}>
-      <Route element={<DevicesPage />} path={getMainRoute()} />
-      <Route element={<DevicesPage />} path={getDevicesRoute()} />
-      <Route element={<ControlPage />} path={getControlRoute(':serial')} />
-      <Route element={<SettingsPage />} path={getSettingsRoute()}>
-        <Route element={<SettingsPage />} path='keys' />
-        <Route element={<SettingsPage />} path='groups' />
-        <Route element={<SettingsPage />} path='devices' />
-        <Route element={<SettingsPage />} path='users' />
-        <Route element={<SettingsPage />} path='shell' />
+    <Route
+      element={
+        <ErrorBoundary FallbackComponent={ErrorFallback}>
```

Comment:
I think we should move the error boundary deeper in the hierarchy to have more control and, for example, not hide the navigation bar on error.  
I think if we have an error that's related to UI instead of the store management - there is no reason to refresh the page and switching to a different page from the UI might restore the application to it's correct state